support/docker/build.sh

```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/env bash
+
+#
+# This script is ran from outside the docker environment, to run itself into
+# the docker environment.
+#
+# It will automatically handle calling `nix-build` correctly, and unwrapping
+# the `result` symlink into a real directory.
+#
+# A `Tow-Boot--nix-store` volume will be automatically created. It will hold
+# the `/nix/store` as a local binary cache. Otherwise the toolchain must be
+# downloaded again every invocation.
+#
+# This is not a one-to-one replacement for Nix, but should be sufficient for
+# basic building locally without installing Nix globally, if Docker is deemed
+# more appropriate.
+#
+
+set -e
+set -u
+set -o pipefail
+PS4=" $ "
+
+_nix_docker() {
+	local STORE_VOLUME
+	STORE_VOLUME="$(docker volume create "Tow-Boot--nix-store")"
+
+	exec docker run \
+		--volume "$STORE_VOLUME:/nix" \
+		--volume "$PWD:/Tow-Boot" \
```

Comment:
This assumes that the user is currently in the Tow-Boot root directory.

```suggestion
		--volume "$(cd "$(dirname "$0")/../.."; pwd):/Tow-Boot" \
```
While this also makes an assumption (that the script hasn't been moved elsewhere), I think that's an assumption that's more likely to hold.